scylla/src/frame/value_tests.rs

```diff
@@ -723,3 +724,156 @@ async fn test_cqlvalue_udt() {
 
     assert_eq!(received_udt_cql_value, &udt_cql_value);
 }
+
+#[tokio::test]
+async fn test_cqlvalue_duration() {
+    let uri = env::var("SCYLLA_URI").unwrap_or_else(|_| "127.0.0.1:9042".to_string());
+    let session: Session = SessionBuilder::new().known_node(uri).build().await.unwrap();
+    let ks = crate::transport::session_test::unique_name();
+    session
+        .query(
+            format!(
+                "CREATE KEYSPACE IF NOT EXISTS {} WITH REPLICATION = \
+            {{'class' : 'SimpleStrategy', 'replication_factor' : 1}}",
+                ks
+            ),
+            &[],
+        )
+        .await
+        .unwrap();
+    session.use_keyspace(&ks, false).await.unwrap();
+
+    session
+        .query(
+            "CREATE TABLE IF NOT EXISTS cqlvalue_duration_test (k int, my duration, primary key (k))",
+            &[],
+        )
+        .await
+        .unwrap();
+
+    let duration_cql_value = CqlValue::Duration(CqlDuration {
+        months: 6,
+        days: 9,
+        nanoseconds: 21372137,
+    });
+
+    session
+        .query(
+            "INSERT INTO cqlvalue_duration_test (k, my) VALUES (5, ?)",
+            (&duration_cql_value,),
+        )
+        .await
+        .unwrap();
+
+    session
```

Comment:
I think we can make this test shorter if you declare a list of strings, loop over them, construct appropriate query strings and `.query` them.